dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java

```diff
@@ -194,8 +207,15 @@ protected NavResultHydrated getNav(Host host, String path, long languageId, User
                     children.add(nav);
                 } else if (item instanceof IFileAsset) {
                     IFileAsset itemFile = (IFileAsset) item;
+                    boolean fileExistInRequestedLanguage = true;
+
+                    if(itemFile.getLanguageId() != languageId){
+                        fileExistInRequestedLanguage = findFileAssetInRequestedLanguage(menuItems,itemFile.getPermissionId(),languageId);
+                    }
+
 
-                    if (itemFile.getLanguageId() == languageId || LanguageWebAPI.canDefaultFileToDefaultLanguage()) {
+                    if (itemFile.getLanguageId() == languageId ||
+                            (LanguageWebAPI.canDefaultFileToDefaultLanguage() && itemFile.getLanguageId() == defaultLanguage && !fileExistInRequestedLanguage)) {
```

Comment:
same as above, extract to little method the second part of the OR condition: shouldAddFileInGivenLanguage(langId)